packages/core/src/odata/v2/request-builder/response-data-accessor.ts

```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+
+export function getCollectionResult(data): any[] | any {
```

Comment:
Minor: I find it strange that the `getCollectionResult` can return `any` and not only collection. I see the reason in the workaround. However I would do the follwoing. make the getCollection return always a collection since this is also exported.

Use for the getSingleResult() the following:
```
export function getSingleResult(data): Record<string, any> {
  return isSingleResultAsCollection(data)
    ? data.d.results
    : data?.d || {};
}
```

Reply:
Yes, I also did that in the beginning and changed it back and fro.